superset/assets/javascripts/dashboard/components/SliceContainer.jsx

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable react/no-danger */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import SliceHeader from './SliceHeader';
+import Slice from './Slice';
+import StackTraceMessage from '../../components/StackTraceMessage';
+import vizMap from '../../../visualizations/main';
+import * as Actions from '../actions';
+import { t } from '../../locales';
+
+import '../../../stylesheets/dashboard.css';
+
+const propTypes = {
+  actions: PropTypes.object.isRequired,
+  formData: PropTypes.object,
+  timeout: PropTypes.number,
+  datasource: PropTypes.object,
+  isLoading: PropTypes.bool,
+  isExpanded: PropTypes.bool,
+  widgetHeight: PropTypes.number,
+  widgetWidth: PropTypes.number,
+  fetchSlice: PropTypes.func,
+  removeSlice: PropTypes.func.isRequired,
+  updateSliceName: PropTypes.func,
+  getFormDataExtra: PropTypes.func,
+  slice: PropTypes.object,
+  filters: PropTypes.object,
+  refresh: PropTypes.bool,
+};
+
+class SliceContainer extends React.PureComponent {
```

Comment:
About naming and modeling of components here, I think ideally we'd have something like 
* `SliceContainer`, that is simple as possible and only does the redux interaction and passes on props to
* `Slice` which represents the whole slice, header and body
* `SliceBody` which is now called `Slice`

Reply:
Please see the latest commit. Now i wrap name component like `ChartContainer`, `Chart`, and `ChartBody`. 
`ChartContainer` is redux container.
`Chart` only responsible for chart rendering part. It's parent component, either dashboard or explore view, provide header controls.
`ChartBody` provides dom operation wrapper called from d3.